From e1c56acf3355cd539447511fdc1b886e5eb5cca3 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 2 May 2024 16:57:31 +0100
Subject: [PATCH 1182/1215] dts: bcm2712: Dedup the aliases and overrides

Move the aliases and overrrides shared by Pi 5 and CM5 into
bcm2712-rpi.dtsi.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 .../boot/dts/broadcom/bcm2712-rpi-5-b.dts     | 115 -----------------
 .../boot/dts/broadcom/bcm2712-rpi-cm5.dtsi    | 114 -----------------
 arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi | 117 +++++++++++++++++-
 3 files changed, 113 insertions(+), 233 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -746,122 +746,7 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 };
 
 / {
-	aliases: aliases {
-		blconfig = &blconfig;
-		blpubkey = &blpubkey;
-		bluetooth = &bluetooth;
-		console = &uart10;
-		ethernet0 = &rp1_eth;
-		wifi0 = &wifi;
-		fb = &fb;
-		mailbox = &mailbox;
-		mmc0 = &sdio1;
-		uart0 = &uart0;
-		uart1 = &uart1;
-		uart2 = &uart2;
-		uart3 = &uart3;
-		uart4 = &uart4;
-		uart10 = &uart10;
-		serial0 = &uart0;
-		serial1 = &uart1;
-		serial2 = &uart2;
-		serial3 = &uart3;
-		serial4 = &uart4;
-		serial10 = &uart10;
-		i2c = &i2c_arm;
-		i2c0 = &i2c0;
-		i2c1 = &i2c1;
-		i2c2 = &i2c2;
-		i2c3 = &i2c3;
-		i2c4 = &i2c4;
-		i2c5 = &i2c5;
-		i2c6 = &i2c6;
-		i2c10 = &i2c_rp1boot;
-		// Bit-bashed i2c_gpios start at 10
-		spi0 = &spi0;
-		spi1 = &spi1;
-		spi2 = &spi2;
-		spi3 = &spi3;
-		spi4 = &spi4;
-		spi5 = &spi5;
-		spi10 = &spi10;
-		gpio0 = &gpio;
-		gpio1 = &gio;
-		gpio2 = &gio_aon;
-		gpio3 = &pinctrl;
-		gpio4 = &pinctrl_aon;
-		usb0 = &rp1_usb0;
-		usb1 = &rp1_usb1;
-		drm-dsi1 = &dsi0;
-		drm-dsi2 = &dsi1;
-	};
-
 	__overrides__ {
-		bdaddr = <&bluetooth>, "local-bd-address[";
-		button_debounce = <&pwr_key>, "debounce-interval:0";
-		cooling_fan = <&fan>, "status", <&rp1_pwm1>, "status";
-		uart0_console = <&uart0>,"status", <&aliases>, "console=",&uart0;
-		i2c0 = <&i2c0>, "status";
-		i2c1 = <&i2c1>, "status";
-		i2c = <&i2c1>, "status";
-		i2c_arm = <&i2c_arm>, "status";
-		i2c_vc = <&i2c_vc>, "status";
-		i2c_csi_dsi = <&i2c_csi_dsi>, "status";
-		i2c_csi_dsi0 = <&i2c_csi_dsi0>, "status";
-		i2c_csi_dsi1 = <&i2c_csi_dsi1>, "status";
-		i2c0_baudrate = <&i2c0>, "clock-frequency:0";
-		i2c1_baudrate = <&i2c1>, "clock-frequency:0";
-		i2c_baudrate = <&i2c_arm>, "clock-frequency:0";
-		i2c_arm_baudrate = <&i2c_arm>, "clock-frequency:0";
-		i2c_vc_baudrate = <&i2c_vc>, "clock-frequency:0";
-		krnbt = <&bluetooth>, "status";
-		nvme = <&pciex1>, "status";
-		pciex1 = <&pciex1>, "status";
-		pciex1_gen = <&pciex1> , "max-link-speed:0";
-		pciex1_no_l0s = <&pciex1>, "aspm-no-l0s?";
-		pciex1_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
-		pcie_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
-		random = <&random>, "status";
-		rtc = <&rpi_rtc>, "status";
-		rtc_bbat_vchg = <&rpi_rtc>, "trickle-charge-microvolt:0";
 		sd_cqe = <&sdio1>, "supports-cqe?";
-		spi = <&spi0>, "status";
-		suspend = <&pwr_key>, "linux,code:0=205";
-		uart0 = <&uart0>, "status";
-		wifiaddr = <&wifi>, "local-mac-address[";
-
-		act_led_gpio = <&led_act>,"gpios:4",<&led_act>,"gpios:0=",<&gpio>;
-		act_led_activelow = <&led_act>,"gpios:8";
-		act_led_trigger = <&led_act>, "linux,default-trigger";
-		pwr_led_gpio = <&led_pwr>,"gpios:4";
-		pwr_led_activelow = <&led_pwr>, "gpios:8";
-		pwr_led_trigger = <&led_pwr>, "linux,default-trigger";
-		eth_led0 = <&phy1>,"led-modes:0";
-		eth_led1 = <&phy1>,"led-modes:4";
-		drm_fb0_rp1_dsi0 = <&aliases>, "drm-fb0=",&dsi0;
-		drm_fb0_rp1_dsi1 = <&aliases>, "drm-fb0=",&dsi1;
-		drm_fb0_rp1_dpi = <&aliases>, "drm-fb0=",&dpi;
-		drm_fb0_vc4 = <&aliases>, "drm-fb0=",&vc4;
-		drm_fb1_rp1_dsi0 = <&aliases>, "drm-fb1=",&dsi0;
-		drm_fb1_rp1_dsi1 = <&aliases>, "drm-fb1=",&dsi1;
-		drm_fb1_rp1_dpi = <&aliases>, "drm-fb1=",&dpi;
-		drm_fb1_vc4 = <&aliases>, "drm-fb1=",&vc4;
-		drm_fb2_rp1_dsi0 = <&aliases>, "drm-fb2=",&dsi0;
-		drm_fb2_rp1_dsi1 = <&aliases>, "drm-fb2=",&dsi1;
-		drm_fb2_rp1_dpi = <&aliases>, "drm-fb2=",&dpi;
-		drm_fb2_vc4 = <&aliases>, "drm-fb2=",&vc4;
-
-		fan_temp0 = <&cpu_tepid>,"temperature:0";
-		fan_temp1 = <&cpu_warm>,"temperature:0";
-		fan_temp2 = <&cpu_hot>,"temperature:0";
-		fan_temp3 = <&cpu_vhot>,"temperature:0";
-		fan_temp0_hyst = <&cpu_tepid>,"hysteresis:0";
-		fan_temp1_hyst = <&cpu_warm>,"hysteresis:0";
-		fan_temp2_hyst = <&cpu_hot>,"hysteresis:0";
-		fan_temp3_hyst = <&cpu_vhot>,"hysteresis:0";
-		fan_temp0_speed = <&fan>, "cooling-levels:4";
-		fan_temp1_speed = <&fan>, "cooling-levels:8";
-		fan_temp2_speed = <&fan>, "cooling-levels:12";
-		fan_temp3_speed = <&fan>, "cooling-levels:16";
 	};
 };
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -753,108 +753,7 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 };
 
 / {
-	aliases: aliases {
-		blconfig = &blconfig;
-		blpubkey = &blpubkey;
-		bluetooth = &bluetooth;
-		console = &uart10;
-		ethernet0 = &rp1_eth;
-		wifi0 = &wifi;
-		fb = &fb;
-		mailbox = &mailbox;
-		mmc0 = &sdio1;
-		uart0 = &uart0;
-		uart1 = &uart1;
-		uart2 = &uart2;
-		uart3 = &uart3;
-		uart4 = &uart4;
-		uart10 = &uart10;
-		serial0 = &uart0;
-		serial1 = &uart1;
-		serial2 = &uart2;
-		serial3 = &uart3;
-		serial4 = &uart4;
-		serial10 = &uart10;
-		i2c = &i2c_arm;
-		i2c0 = &i2c0;
-		i2c1 = &i2c1;
-		i2c2 = &i2c2;
-		i2c3 = &i2c3;
-		i2c4 = &i2c4;
-		i2c5 = &i2c5;
-		i2c6 = &i2c6;
-		i2c10 = &i2c_rp1boot;
-		// Bit-bashed i2c_gpios start at 10
-		spi0 = &spi0;
-		spi1 = &spi1;
-		spi2 = &spi2;
-		spi3 = &spi3;
-		spi4 = &spi4;
-		spi5 = &spi5;
-		spi10 = &spi10;
-		gpio0 = &gpio;
-		gpio1 = &gio;
-		gpio2 = &gio_aon;
-		gpio3 = &pinctrl;
-		gpio4 = &pinctrl_aon;
-		usb0 = &rp1_usb0;
-		usb1 = &rp1_usb1;
-		drm-dsi1 = &dsi0;
-		drm-dsi2 = &dsi1;
-	};
-
 	__overrides__ {
-		bdaddr = <&bluetooth>, "local-bd-address[";
-		button_debounce = <&pwr_key>, "debounce-interval:0";
-		cooling_fan = <&fan>, "status", <&rp1_pwm1>, "status";
-		uart0_console = <&uart0>,"status", <&aliases>, "console=",&uart0;
-		i2c0 = <&i2c0>, "status";
-		i2c1 = <&i2c1>, "status";
-		i2c = <&i2c1>, "status";
-		i2c_arm = <&i2c_arm>, "status";
-		i2c_vc = <&i2c_vc>, "status";
-		i2c_csi_dsi = <&i2c_csi_dsi>, "status";
-		i2c_csi_dsi0 = <&i2c_csi_dsi0>, "status";
-		i2c_csi_dsi1 = <&i2c_csi_dsi1>, "status";
-		i2c0_baudrate = <&i2c0>, "clock-frequency:0";
-		i2c1_baudrate = <&i2c1>, "clock-frequency:0";
-		i2c_baudrate = <&i2c_arm>, "clock-frequency:0";
-		i2c_arm_baudrate = <&i2c_arm>, "clock-frequency:0";
-		i2c_vc_baudrate = <&i2c_vc>, "clock-frequency:0";
-		krnbt = <&bluetooth>, "status";
-		nvme = <&pciex1>, "status";
-		pciex1 = <&pciex1>, "status";
-		pciex1_gen = <&pciex1> , "max-link-speed:0";
-		pciex1_no_l0s = <&pciex1>, "aspm-no-l0s?";
-		pciex1_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
-		pcie_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
-		random = <&random>, "status";
-		rtc = <&rpi_rtc>, "status";
-		rtc_bbat_vchg = <&rpi_rtc>, "trickle-charge-microvolt:0";
-		spi = <&spi0>, "status";
-		suspend = <&pwr_key>, "linux,code:0=205";
-		uart0 = <&uart0>, "status";
-		wifiaddr = <&wifi>, "local-mac-address[";
-
-		act_led_activelow = <&led_act>, "active-low?";
-		act_led_trigger = <&led_act>, "linux,default-trigger";
-		pwr_led_activelow = <&led_pwr>, "gpios:8";
-		pwr_led_trigger = <&led_pwr>, "linux,default-trigger";
-		eth_led0 = <&phy1>,"led-modes:0";
-		eth_led1 = <&phy1>,"led-modes:4";
-		drm_fb0_rp1_dsi0 = <&aliases>, "drm-fb0=",&dsi0;
-		drm_fb0_rp1_dsi1 = <&aliases>, "drm-fb0=",&dsi1;
-		drm_fb0_rp1_dpi = <&aliases>, "drm-fb0=",&dpi;
-		drm_fb0_vc4 = <&aliases>, "drm-fb0=",&vc4;
-		drm_fb1_rp1_dsi0 = <&aliases>, "drm-fb1=",&dsi0;
-		drm_fb1_rp1_dsi1 = <&aliases>, "drm-fb1=",&dsi1;
-		drm_fb1_rp1_dpi = <&aliases>, "drm-fb1=",&dpi;
-		drm_fb1_vc4 = <&aliases>, "drm-fb1=",&vc4;
-		drm_fb2_rp1_dsi0 = <&aliases>, "drm-fb2=",&dsi0;
-		drm_fb2_rp1_dsi1 = <&aliases>, "drm-fb2=",&dsi1;
-		drm_fb2_rp1_dpi = <&aliases>, "drm-fb2=",&dpi;
-		drm_fb2_vc4 = <&aliases>, "drm-fb2=",&vc4;
-
 		ant1 =  <&ant1>,"output-high?=on",
 			<&ant1>, "output-low?=off",
 			<&ant2>, "output-high?=off",
@@ -867,18 +766,5 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 			<&ant1>, "output-low?=on",
 			<&ant2>, "output-high?=off",
 			<&ant2>, "output-low?=on";
-
-		fan_temp0 = <&cpu_tepid>,"temperature:0";
-		fan_temp1 = <&cpu_warm>,"temperature:0";
-		fan_temp2 = <&cpu_hot>,"temperature:0";
-		fan_temp3 = <&cpu_vhot>,"temperature:0";
-		fan_temp0_hyst = <&cpu_tepid>,"hysteresis:0";
-		fan_temp1_hyst = <&cpu_warm>,"hysteresis:0";
-		fan_temp2_hyst = <&cpu_hot>,"hysteresis:0";
-		fan_temp3_hyst = <&cpu_vhot>,"hysteresis:0";
-		fan_temp0_speed = <&fan>, "cooling-levels:4";
-		fan_temp1_speed = <&fan>, "cooling-levels:8";
-		fan_temp2_speed = <&fan>, "cooling-levels:12";
-		fan_temp3_speed = <&fan>, "cooling-levels:16";
 	};
 };
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -98,14 +98,124 @@
 };
 
 / {
+	aliases: aliases {
+		blconfig = &blconfig;
+		blpubkey = &blpubkey;
+		bluetooth = &bluetooth;
+		console = &uart10;
+		drm-dsi1 = &dsi0;
+		drm-dsi2 = &dsi1;
+		ethernet0 = &rp1_eth;
+		fb = &fb;
+		gpio0 = &gpio;
+		gpio1 = &gio;
+		gpio2 = &gio_aon;
+		gpio3 = &pinctrl;
+		gpio4 = &pinctrl_aon;
+		i2c = &i2c_arm;
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		i2c10 = &i2c_rp1boot;
+		i2c2 = &i2c2;
+		i2c3 = &i2c3;
+		i2c4 = &i2c4;
+		i2c5 = &i2c5;
+		i2c6 = &i2c6;
+		mailbox = &mailbox;
+		mmc0 = &sdio1;
+		serial0 = &uart0;
+		serial1 = &uart1;
+		serial10 = &uart10;
+		serial2 = &uart2;
+		serial3 = &uart3;
+		serial4 = &uart4;
+		spi0 = &spi0;
+		spi1 = &spi1;
+		spi10 = &spi10;
+		spi2 = &spi2;
+		spi3 = &spi3;
+		spi4 = &spi4;
+		spi5 = &spi5;
+		uart0 = &uart0;
+		uart1 = &uart1;
+		uart10 = &uart10;
+		uart2 = &uart2;
+		uart3 = &uart3;
+		uart4 = &uart4;
+		usb0 = &rp1_usb0;
+		usb1 = &rp1_usb1;
+		wifi0 = &wifi;
+	};
+
 	__overrides__ {
-		arm_freq;
+		act_led_gpio = <&led_act>,"gpios:4",<&led_act>,"gpios:0=",<&gpio>;
+		act_led_activelow = <&led_act>, "gpios:8";
+		act_led_trigger = <&led_act>, "linux,default-trigger";
 		axiperf = <&axiperf>,"status";
-
+		bdaddr = <&bluetooth>, "local-bd-address[";
+		button_debounce = <&pwr_key>, "debounce-interval:0";
+		cooling_fan = <&fan>, "status", <&rp1_pwm1>, "status";
+		drm_fb0_rp1_dpi = <&aliases>, "drm-fb0=",&dpi;
+		drm_fb0_rp1_dsi0 = <&aliases>, "drm-fb0=",&dsi0;
+		drm_fb0_rp1_dsi1 = <&aliases>, "drm-fb0=",&dsi1;
+		drm_fb0_vc4 = <&aliases>, "drm-fb0=",&vc4;
+		drm_fb1_rp1_dpi = <&aliases>, "drm-fb1=",&dpi;
+		drm_fb1_rp1_dsi0 = <&aliases>, "drm-fb1=",&dsi0;
+		drm_fb1_rp1_dsi1 = <&aliases>, "drm-fb1=",&dsi1;
+		drm_fb1_vc4 = <&aliases>, "drm-fb1=",&vc4;
+		drm_fb2_rp1_dpi = <&aliases>, "drm-fb2=",&dpi;
+		drm_fb2_rp1_dsi0 = <&aliases>, "drm-fb2=",&dsi0;
+		drm_fb2_rp1_dsi1 = <&aliases>, "drm-fb2=",&dsi1;
+		drm_fb2_vc4 = <&aliases>, "drm-fb2=",&vc4;
+		eth_led0 = <&phy1>,"led-modes:0";
+		eth_led1 = <&phy1>,"led-modes:4";
+		fan_temp0 = <&cpu_tepid>,"temperature:0";
+		fan_temp0_hyst = <&cpu_tepid>,"hysteresis:0";
+		fan_temp0_speed = <&fan>, "cooling-levels:4";
+		fan_temp1 = <&cpu_warm>,"temperature:0";
+		fan_temp1_hyst = <&cpu_warm>,"hysteresis:0";
+		fan_temp1_speed = <&fan>, "cooling-levels:8";
+		fan_temp2 = <&cpu_hot>,"temperature:0";
+		fan_temp2_hyst = <&cpu_hot>,"hysteresis:0";
+		fan_temp2_speed = <&fan>, "cooling-levels:12";
+		fan_temp3 = <&cpu_vhot>,"temperature:0";
+		fan_temp3_hyst = <&cpu_vhot>,"hysteresis:0";
+		fan_temp3_speed = <&fan>, "cooling-levels:16";
+		i2c = <&i2c1>, "status";
+		i2c_arm = <&i2c_arm>, "status";
+		i2c_arm_baudrate = <&i2c_arm>, "clock-frequency:0";
+		i2c_baudrate = <&i2c_arm>, "clock-frequency:0";
+		i2c_csi_dsi = <&i2c_csi_dsi>, "status";
+		i2c_csi_dsi0 = <&i2c_csi_dsi0>, "status";
+		i2c_csi_dsi1 = <&i2c_csi_dsi1>, "status";
+		i2c_vc = <&i2c_vc>, "status";
+		i2c_vc_baudrate = <&i2c_vc>, "clock-frequency:0";
+		i2c0 = <&i2c0>, "status";
+		i2c0_baudrate = <&i2c0>, "clock-frequency:0";
+		i2c1 = <&i2c1>, "status";
+		i2c1_baudrate = <&i2c1>, "clock-frequency:0";
+		krnbt = <&bluetooth>, "status";
+		nvme = <&pciex1>, "status";
 		nvmem_cust_rw = <&nvmem_cust>,"rw?";
-		nvmem_priv_rw = <&nvmem_priv>,"rw?";
 		nvmem_mac_rw = <&nvmem_mac>,"rw?";
+		nvmem_priv_rw = <&nvmem_priv>,"rw?";
+		pcie_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
+		pciex1 = <&pciex1>, "status";
+		pciex1_gen = <&pciex1> , "max-link-speed:0";
+		pciex1_no_l0s = <&pciex1>, "aspm-no-l0s?";
+		pciex1_tperst_clk_ms = <&pciex1>, "brcm,tperst-clk-ms:0";
+		pwr_led_gpio = <&led_pwr>, "gpios:4";
+		pwr_led_activelow = <&led_pwr>, "gpios:8";
+		pwr_led_trigger = <&led_pwr>, "linux,default-trigger";
+		random = <&random>, "status";
+		rtc = <&rpi_rtc>, "status";
+		rtc_bbat_vchg = <&rpi_rtc>, "trickle-charge-microvolt:0";
+		spi = <&spi0>, "status";
 		strict_gpiod = <&chosen>, "bootargs=pinctrl_rp1.persist_gpio_outputs=n";
+		suspend = <&pwr_key>, "linux,code:0=205";
+		uart0 = <&uart0>, "status";
+		uart0_console = <&uart0>,"status", <&aliases>, "console=",&uart0;
+		wifiaddr = <&wifi>, "local-mac-address[";
 
 		cam0_reg = <&cam0_reg>,"status";
 		cam0_reg_gpio = <&cam0_reg>,"gpio:4",
@@ -113,7 +223,6 @@
 		cam1_reg = <&cam1_reg>,"status";
 		cam1_reg_gpio = <&cam1_reg>,"gpio:4",
 				<&cam1_reg>,"gpio:0=", <&gpio>;
-
 	};
 };
 
